datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -950,6 +950,24 @@ macro_rules! assert_contains {
     };
 }
 
+/// Apply simplification and constant propagation to ([Expr]).
+///
+/// # Arguments
+///
+/// * `expr` - The logical expression
+/// * `schema` - The DataFusion schema for the expr, used to resolve `Column` references
+///                      to qualified or unqualified fields by name.
+/// * `props` - The Arrow schema for the input, used for determining expression data types
+///                    when performing type coercion.
+pub fn simplify_expr(
+    expr: Expr,
+    schema: DFSchemaRef,
```

Comment:
Since `SimplifyContext::new` needs a `Vec<&'Arc<DFSchema>`, so I change the input `schema`  to `Arc<DFSchema>` to avoid clone.

Reply:
I think this makes sense -- the SimplifyContext constructor is somewhat akward

Reply:
While working on https://github.com/apache/arrow-datafusion/pull/3741 I found having a signature of `schema: &DFSchemaRef` was actually more ergnonmic. What do you think?

We have to rerun the CI anyways give the github nonsense earlier today

Reply:
Addressed.